packages/synchro-charts/src/components/sc-dial/sc-dial-base/sc-dial-base.tsx

```diff
@@ -0,0 +1,186 @@
+import { Component, h, Prop, State, Watch } from '@stencil/core';
+import { DataPoint, DataStream, MinimalSizeConfig, Primitive, ViewPortConfig } from '../../../utils/dataTypes';
+import { round } from '../../../utils/number';
+import { isNumberDataStream } from '../../../utils/predicates';
+import { getIcons } from '../../charts/common/annotations/iconUtils';
+import { Threshold } from '../../charts/common/types';
+import { NO_VALUE_PRESENT } from '../../common/terms';
+import { DialLoading } from './sc-dial-loading';
+import { TextSizeConfig, sizeContent, StatusProgress, LengthToSize, SizeFont } from './util';
+
+const title = (dataStream: { detailedName?: any; name?: any } | null | false) => {
+  if (dataStream) {
+    return dataStream.detailedName || dataStream.name;
+  }
+  return null;
+};
+
+const UNITMINLENGTH = 1;
+@Component({
+  tag: 'sc-dial-base',
+  styleUrl: 'sc-dial-base.css',
+  shadow: false,
+})
+export class ScDialBase {
+  @Prop() viewport: ViewPortConfig;
+  @Prop() breachedThreshold?: Threshold;
+  @Prop() valueColor?: string; // css color string
+
+  @Prop() alarmStream?: DataStream;
+
+  @Prop() propertyStream?: DataStream;
+  @Prop() propertyPoint?: DataPoint<Primitive>;
+  @Prop() size?: MinimalSizeConfig = { width: 0, height: 0 };
+
+  @Prop() isLoading?: boolean = false;
+
+  @State() fontSize: TextSizeConfig = sizeContent.XL;
```

Comment:
The size should have the ability to be specified explicitly at the top level, i.e. as a user, I should be able to say

```
<sc-dial
  ...
  size="XL"
/>
```

And have a dial which is large

Reply:
We've defined the width and height with size, and I add fontSize property into size object. And there are still some things I do not understand. We need to have a meeting on Wednesday to communicate.